### ISSUE TITLE
RemoteExporter in SpringBoot 3.0.0

### ISSUE THREAD

**Issue by b-artur** [briandilley/jsonrpc4j#303]
Unfortunately application with jsonrpc4j fails on run in a new release of SpringBoot 3.0.0
```
ERROR 22396 --- [           main] o.s.boot.SpringApplication               : Application run failed

java.lang.NoClassDefFoundError: org/springframework/remoting/support/RemoteExporter
```

**Comment by cyb3r4nt** [comment]
New major Spring Framework version 6 has removed the `RemoteExporter` class and RPC-style remoting:

https://github.com/spring-projects/spring-framework/issues/25379#issuecomment-656228589
https://github.com/spring-projects/spring-framework/issues/27422

It is still possible to use this library without Spring integration and by creating  the `JsonRpcHttpClient` and `JsonRpcServer` objects manually,
and propagating them as beans to application context.
https://github.com/briandilley/jsonrpc4j#without-the-spring-framework

Then you can use the library to create and handle requests.
To use  `JsonRpcServer`  you may create explicit Spring MVC controllers,
and redirect request handling to `jsonrpc4j`

```java
@Controller
class SZv2JsonRpcController {

    private final JsonRpcServer myJsonRpcServer;

    SZv2JsonRpcController(JsonRpcServer myJsonRpcServer) {
        this.myJsonRpcServer = myJsonRpcServer;
    }

    @PostMapping(path = "/api")
    public void handleMyRequest(
        jakarta.servlet.http.HttpServletRequest req,
        jakarta.servlet.http.HttpServletResponse resp
    ) throws IOException {
        this.myJsonRpcServer.handle(
            new Jakarta2JavaxHttpServletRequestAdapter(req),
            new Jakarta2JavaxHttpServletResponseAdapter(resp)
        );
    }
}
```
The used API is https://github.com/briandilley/jsonrpc4j/blob/master/src/main/java/com/googlecode/jsonrpc4j/JsonRpcServer.java#L107-L140
There is one more obstacle because of `javax` -> `jakarta` namespace migration in the newer Spring Framework,
and you will need to create some adapter classes.

But this is a working solution you can use until better solution is introduced into `jsonrpc4j`

**Comment by nixel2007** [comment]
@cyb3r4nt maybe you have a complete example for spring boot 6?

**Comment by cyb3r4nt** [comment]
> @cyb3r4nt maybe you have a complete example for spring boot 6?

The example provided above works with Spring Framework 6.
The only thing is that it does not use the Spring integration from `jsonrpc4j`,
and you need to define required controllers manually and route requests to proper `JsonRpcServer`.
It is possible to register required beans in the application context ,
and define `JsonRpcServer` implementation beans and corresponding Spring MVC controllers.
Such routing requires some more wiring code, but it may solve current problems.

Spring Framework 6 integration in `jsonrpc4j` is broken at the moment, because when you use it,
then you will get `NoClassDefFoundError` because  `RemoteExporter` is no more in the classpath.
`@JsonRpcService`  and `@AutoJsonRpcServiceImpl` auto-discovery annotations cannot be used because of  `RemoteExporter` usage in the `AbstractJsonServiceExporter`.

(PS Spring Boot and Spring Framework are different projects, and they have different versioning. Please do not mix them)

**Comment by cyb3r4nt** [comment]
Also, just checked the `RemoteExporter` usage and how beans are registered in the `AutoJsonRpcServiceImplExporter`.
At very first look it seems that `RemoteExporter`  is used only because of `serviceInterface` , `service` and `proxyForService` class properties,
and uses them only to get the request handler object  when creating `JsonRpcServer` instance.

Otherwise, all created `JsonRpcServer` instances are registered in the  Spring IoC container using bean-name-as-web-path scheme (see `BeanNameUrlHandlerMapping` for more details) and this is more or less standard way how Spring Web works.
It also seems that `org.springframework.remoting` is not really used  and there are no any RPC-like registrations.

Those properties used in `AutoJsonRpcServiceImplExporter` may be pulled down from the `RemoteExporter` and this inheritance may be removed/discarded.

Will try to develop and provide some patches in spare time.

**Comment by reneleonhardt** [comment]
@b-artur the PR has been merged and a new version 1.7 has been released in case you still need it 😄

**Comment by Zurcusa** [comment]
@briandilley Is there a plan to migrate the project to the `jakarta` namespace in the future? I saw there is an option to create `jakarta` to `javax` adapters and use them together with the `JsonRpcServer`, but I'd like to avoid adding the `javax` dependencies.

**Comment by reneleonhardt** [comment]
If upgrades would be allowed now, Spring 6 already has a Jakarta EE 9 baseline:
https://github.com/briandilley/jsonrpc4j/issues/330#issuecomment-2869120897
https://github.com/spring-projects/spring-framework/wiki/Spring-Framework-Versions

**Comment by Zurcusa** [comment]
Yup. I know, that's why we're having issues with using the new version. There's an existing [fork](https://github.com/LimeChain/jsonrpc4j) that works well with Spring and jakarta, but it hasn't been updated with the newly added PRs.

**Comment by reneleonhardt** [comment]
Unfortunately it's not uncommon for projects and their forks to become less and less maintained over time, even when I suggest free AI help they don't want to try CodeRabbit to save contributors and maintainers time, and big companies contribute not enough back or even pay engineers for their globally appreciated efforts 😞

But Brian announced to resume work or at least maintainership on this project, maybe he will start accepting new PRs soon.
